src/components/Menu/Menu.tsx

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2023 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { FC, ReactNode, useMemo } from "react";
+import { Root, Trigger, Portal, Content } from "@radix-ui/react-dropdown-menu";
+import { FloatingMenu } from "./FloatingMenu";
+import { Drawer } from "vaul";
+import classnames from "classnames";
+import drawerMenu from "./DrawerMenu.module.css";
+import { MenuContext } from "./MenuContext";
+import { DrawerMenu } from "./DrawerMenu";
+import { useTouchscreen } from "../../utils/useTouchscreen";
+
+interface Props {
+  /**
+   * The menu title.
+   */
+  title: string;
+  /**
+   * Whether the menu is open.
+   */
+  open: boolean;
```

Comment:
We shouldn't have to pass that prop most of the time, as `trigger` is the main thing that will control the open state of the menu?

```suggestion
  open?: boolean;
```

Reply:
I think your reasoning is totally fine, let's go with that. We might need to do a similar exercise to what Quentin did with forms and formalise and appraoch to controlled vs uncontrolled components.

Being opiniated, hence promoting more consistency throughout the codebase

Reply:
> when components are consistently designed to always be controlled rather than uncontrolled, this leads to more obvious code

In my experience this is true, and makes things more testable and predictable.

I created this issue to track this discussion https://github.com/vector-im/compound/issues/268